src/SFA.DAS.ApprenticeCommitments.Web/Pages/ConfirmYourIdentity.cshtml.cs

```diff
@@ -60,8 +61,17 @@ public class ConfirmYourIdentityModel : PageModel
             return Page();
         }
 
-        public IActionResult OnPost()
+        public async Task<IActionResult> OnPost()
         {
+            Guid.TryParse(User.Claims.FirstOrDefault(c => c.Type == "sub")?.Value ?? "", out var regId);
+            await _registrations.Validate(new VerifyRegistrationCommand
+            {
+                RegistrationId = regId,
+                FirstName = FirstName,
+                LastName = LastName,
+                NationalInsuranceNumber = NationalInsuranceNumber,
+            });
+
             return RedirectToPage("/ConfirmYourIdentity", new
```

Comment:
If the validation is successful, should this not go to a Home page. And when testing this action in the acceptance tests, not the actual response we get back as that will be outside of our testing scope